src/Uno.UI/UI/Xaml/UIElement.macOS.cs

```diff
@@ -254,17 +272,112 @@ public override void MouseDown(NSEvent evt)
 						OriginalSource = this
 					};
 
-					pointerEventIsHandledInManaged = RaiseEvent(PointerEnteredEvent, args);
+					pointerEventIsHandledInManaged = RaisePointerEvent(PointerPressedEvent, args);
+				}
 
-					args.Handled = false; // reset for "pressed" event
+				if (!pointerEventIsHandledInManaged)
+				{
+					// Bubble up the event natively
+					base.MouseDown(evt);
+				}
+			}
+			catch (Exception e)
+			{
+				Application.Current.RaiseRecoverableUnhandledException(e);
+			}
+		}
+
+		public override void MouseMoved(NSEvent evt)
+		{
+			try
+			{
+				var pointerEventIsHandledInManaged = false;
 
-					pointerEventIsHandledInManaged = RaiseEvent(PointerPressedEvent, args) || pointerEventIsHandledInManaged;
+				if (evt.IsTouchInView(this))
+				{
+					IsPointerPressed = true;
+					IsPointerOver = true;
+
+					// evt.AllTouches raises a invalid selector exception
+					var args = new PointerRoutedEventArgs(null, evt)
+					{
+						CanBubbleNatively = true,
```

Comment:
You can remove this as it set by the ctor of `PointerRoutedEventArgs` (same for all `MouseXXX` overrides :) )

Reply:
Done